### DISCUSSION TITLE
Are TableProperties and/or TableCellProperties included?

### DISCUSSION THREAD

**Discussion by angieherrera** [craftcms/ckeditor#115] [Q&A] [answered]
The predefined build of CKEditor (per their docs) doesn't include TableProperties, TableCellProperties, TableColumnResize, or TableCaption.
![CleanShot 2023-07-10 at 11 47 46](https://github.com/craftcms/ckeditor/assets/442434/993a0cd8-53be-4cd9-83d8-f3d7527ae6de)

It doesn't seem like the Craft build of CKEditor does either? Specifically, I've been trying to figure out how to add TableCellProperties and haven't been able to get it to work (not sure if that's due to it just not being possible or if I'm doing it wrong).

In my particular use case, the client (a bank) needs to be able to vertically align cells as different cells may have different amounts/lengths of content. The default middle align doesn't look great in some scenarios.

**Comment by i-just** [comment, marked as answer]
`TableProperties`, `TableCellProperties`, `TableColumnResize` are not currently included in the Craft’s build of CKEditor, but `TableCaption` is. 

To have `TableCaption` button show (along with the default table row, table column and merge buttons), you need the following in your config:
```
"table": {
  "contentToolbar": [
    "toggleTableCaption",
    "tableRow",
    "tableColumn",
    "mergeTableCells"
  ]
}
```

I just raised a PR (https://github.com/craftcms/ckeditor/pull/117) to add `TableProperties`, `TableCellProperties` too. As with `TableCaption`, they won’t show by default, but you’ll be able to include them by adding `"tableProperties", "tableCellProperties"` to the `table` > `contentToolbar` config.

**Reply by brandonkelly** [reply]
CKEditor 3.5.0 is out with these plugins :tada:

Note that the update requires [Craft 4.5.0](https://craftcms.com/blog/craft-4-5-released).

**Reply by AimeeHendrycks** [reply]
Is there any update on including support for TableColumnResize?

**Reply by twitcher07** [reply]
Yeah I agree. It would be nice to have the TableColumnResize (if not all of the other @ckeditor/ckeditor5-table features) be an option if it's going to be in the default build for the Craft plugin.
